### PR TITLE
[Fixtures] Updated CommonBuildSettings with Xcode 8.3

### PR DESCRIPTION
**Pull request by louisdh**
Notable changes from Xcode 8.2.1 to 8.3:

- `CLANG_WARN_DOCUMENTATION_COMMENTS`, `CLANG_WARN_INFINITE_RECURSION` and `CLANG_WARN_SUSPICIOUS_MOVE ` are now in the base config.
- `CLANG_ANALYZER_NUMBER_OBJECT_CONVERSION` is now `YES_AGGRESSIVE` by default for newly created projects.

### CODE REVIEW

**Comment by louisdh** [comment]
@benasher44 Can you review this please?

File: lib/xcodeproj/constants.rb [hunk 1]
```diff
@@ -337,6 +338,7 @@ module Constants
         'GCC_WARN_UNINITIALIZED_AUTOS'       => 'YES',
         'GCC_WARN_UNUSED_FUNCTION'           => 'YES',
         'GCC_WARN_UNUSED_VARIABLE'           => 'YES',
+        'CLANG_ANALYZER_NUMBER_OBJECT_CONVERSION' => 'YES',
```
Comment (segiddins) [inline lib/xcodeproj/constants.rb]: Please keep the build settings sorted

**Comment by benasher44** [comment]
Thanks again @louisdh

**Comment by louisdh** [comment]
@benasher44 Is a new Xcodeproj release on the horizon? (current one is from [December](https://github.com/CocoaPods/Xcodeproj/releases/tag/1.4.2)). Since CocoaPods uses this as a vital dependency, the updated build settings (such as Whole Module Optimization for Swift) can be of great advantage.

**Comment by benasher44** [comment]
@louisdh I believe we're trying to get this into the 1.2.1 CocoaPods release. These updates I think should be available in the upcoming 1.2.1.rc.1 release specifically.

**Pull request merged**